x-pack/plugins/security_solution/server/endpoint/lib/artifacts/migrate_artifacts_to_fleet.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectsClient, Logger } from 'kibana/server';
+import { EndpointArtifactClientInterface } from '../../services';
+import { InternalArtifactCreateSchema } from '../../schemas';
+import { ArtifactConstants } from './common';
+
+/**
+ * With v7.13, artifact storage was moved from a security_solution saved object to a fleet index
+ * in order to support Fleet Server.
+ */
+export const migrateArtifactsToFleet = async (
+  soClient: SavedObjectsClient,
+  endpointArtifactClient: EndpointArtifactClientInterface,
+  logger: Logger,
+  isFleetServerEnabled: boolean
+): Promise<void> => {
+  if (!isFleetServerEnabled) {
+    logger.info('Skipping Artifacts migration to fleet. [fleetServerEnabled] flag is off');
+    return;
+  }
+
+  let totalArtifactsMigrated = -1;
+  let hasMore = true;
+
+  try {
+    while (hasMore) {
+      // Retrieve list of artifact records
+      const {
+        saved_objects: artifactList,
+        total,
+      } = await soClient.find<InternalArtifactCreateSchema>({
+        type: ArtifactConstants.SAVED_OBJECT_TYPE,
+        page: 1,
+        perPage: 10,
```

Comment:
I'm still wondering if this is too many. Artifacts could be large (even though we are in beta) and this call to elasticsearch will return the encoded (zip) artifact in base64 format.

@scunningham thoughts? should I lower the number of records I retrieve?

Reply:
This should be a one time operation right?  We're just migrating users who have already tried out Endpoint Security, correct?

Reply:
@kevinlog yes and no :) 
yes, items will only be migrated once, but this code (`migrateArtifactstoFleet()`) will be called everytime Kibana is bounced, so it will run, but if the prior time this ran was successful, then there should be no artifacts in the SO and thus nothing else will be migrated.